liferay-echo/liferay-portal
# LPS-166427 | LPS-166622 New related assets collection provider - assets with categories in the same vocabularies

## Motivation

Related tickets
[Link to task](https://issues.liferay.com/browse/LPS-166427)
[Link to bug](https://issues.liferay.com/browse/LPS-166622)

Echo Team is trying to add features based on clients needs. One of them is improving the related collections that are shown in collection fragments for display pages.

One feature that is heavily demanded is to show all related items that share vocabularies, which is what this PR tries to solve.

When working on the task, a bug was found were unavailable collections could be selected. This PR contains the fix, to ensure that the corresponding collections works correctly


## Change summary for LPS-166622 Inactive related collection providers are shown in the UI:

* Filter out collections that are not available
* Add integration tests for available and unavailable collections

## How to test the changes

* Start the portal with FF LPS-166036 disabled
* Create a new Display Page Template for Web Content
* Add a collection fragment to the page
* Click on the collection fragment to configure it
* Go to the related collection tabs
* Assets with categories in the same vocabularies collection provider should not be shown



## Change summary for LPS-166427 New related assets collection provider - assets with categories in the same vocabularies:

* Add new related collection provider to show items that share the same vocabularies as the one that is shown in the display page
* Add integration tests


## How to test the changes

* Start the portal with FF LPS-166036 enabled
* Create a new Display Page Template for Web Content
* Add a collection fragment to the page
* Click on the collection fragment to configure it
* Go to the related collection tabs
* Select Assets with categories in the same vocabularies collection provider
* Add a heading to the collection fragment and map the title of the web content to it
* Publish the display page template and mark it as default
* Add a new vocabulary
* Add 2 categories to the vocabulary
* Create a new web content and assign the first category to it
* Create a second web content and assign the second category to it
* Click on the kebab of the first web content and select "View Content"
* You will see the title of the second web in the list


![related](https://user-images.githubusercontent.com/4267448/197741359-8352ae89-d582-4b7a-b512-068c8e22e2b8.gif)
